projects/ADuCM3029_demo_cn0398/src/ADuCM3029_demo_cn0398.cpp

```diff
@@ -0,0 +1,403 @@
+static void InitSystem(void);
+static void Trap(void);
+static void RepairConnection(void);
+
+
+int main(int argc, char *argv[])
+{
+    CN0398         	cn0398;
+//    SoilMoisture	*pCn = &cn0398;
+    SENSOR_RESULT   eSensorResult;
+
+	/**
+	 * Initialize managed drivers and/or services that have been added to 
+	 * the project.
+	 * @return zero on success 
+	 */
+	
+    /* Initialize the system */
+    InitSystem();
+
+    /* Open Accelerometer */
+    eSensorResult = cn0398.open();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        Trap();
+    }
+
+    /* Start measurement */
+    eSensorResult = cn0398.start();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        Trap();
+    }
+
+	/* Publish ADXL362 accelerometer data using MQTT */
+	MQTTPublishCN0398(cn0398);
+
+	return 0;
+}
+
+/*!
+ * @brief      Application Callback
+ *
+ * @details    Called by the framework layer (adi_wifi_noos.c) when an event occurs.
+ *
+ * @param [in] pCBParam : Callback parameter (unused)
+ *
+ * @param [in] Event : Event of type #ADI_WIFI_AT_CMDCODE.
+ *
+ * @param [in] pArg : Callback argument (unused)
+ *
+ */
+void adi_wifi_AplicationCallback(void * pCBParam, uint32_t nEvent, void * pArg)
+{
+	switch (nEvent)
+		{
+			case CMD_NONE:
+				/* Read the publish Data. */
+				adi_wifi_ParseSubscriberData();
+				break;
+			case CMD_HW_ERROR:
+			{
+			    break;
+			}
+
+			default:
+				break;
+		}
+
+	return;
+}
+
+static void MQTTPublishCN0398(CN0398 pCn)
+{
+	ADI_WIFI_RESULT eResult;
+    uint32_t        nTime = 0u;
+    uint32_t        nPacketId = 0u;
+    uint8_t         nConnectionFailFlag;
+    uint32_t		p_byte;
+
+    /* Initialize the ESP8266 Wi-Fi module and establish a TCP connection */
+    InitWiFiConnection();
+
+    /* Send MQTT "CONNECT" message to broker */
+    eResult = adi_wifi_radio_MQTTConnect(&gMQTTConnect);
+    DEBUG_RESULT("Error connecting to the MQTT broker.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+    /* Start the ping timer to ping the MQTT broker before the keep alive expires */
+    if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) == ADI_WIFI_SUCCESS)
+    {
+    	DEBUG_MESSAGE("Starting to Publish\n");
+
+		gPublishConfig.nPacketId = nPacketId;
+
+		while(1)
+		{
+			nConnectionFailFlag = 0;
+
+            /* If ping timer has expired */
+            if (adi_wifi_IsTimerDone(1u) == 1u)
+            {
+        		/* Send MQTT "PINGREQ" message to broker to keep the connection alive. */
+        		eResult = adi_wifi_radio_MQTTPing(ADI_WIFI_CONNECTION_ID);
+
+        		/* Restart the ping timer */
+        		if((adi_wifi_StopTimer(1u) == ADI_WIFI_SUCCESS) && (eResult == ADI_WIFI_SUCCESS))
+            	{
+        			if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) != ADI_WIFI_SUCCESS)
+                    {
+        		    	DEBUG_MESSAGE("Failed to start ping timer\n");
+
+                    }
```

Comment:
Do not unnecessarily use braces where a single statement will do: `if(adi_wifi_StartTimer(ADI_WIFI_MQTT_PING_TIMEOUT, 1u) != ADI_WIFI_SUCCESS)
                DEBUG_MESSAGE("Failed to start ping timer\n");`